data-tck/src/main/java/io/micronaut/data/tck/repositories/AuthorRepository.java

```diff
@@ -78,15 +79,24 @@ public interface AuthorRepository extends CrudRepository<Author, Long> {
     @Join("books")
     List<Author> listAll();
 
+    @Join("books")
+    List<AuthorDtoWithBooks> queryAll();
+
     @Join(value = "books", type = Join.Type.LEFT_FETCH)
     List<Author> findByIdIsNotNull();
 
+    @Join(value = "books", type = Join.Type.LEFT_FETCH)
```

Comment:
What happens if the join is not specified?

Reply:
The join type or just any join?

Reply:
Added a test